torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -475,12 +475,17 @@ def shard_parameter(
     # Scatter the shards (use broadcast since NCCL doesn't support scatter, this is very inefficient).
     dist.broadcast(tensor, src=src_rank, group=pg)
 
-    # Reshape to get shard for this rank.
-    local_shard = tensor.narrow(
-        sharding_spec.dim,  # type: ignore[arg-type]
-        local_metadata.shard_offsets[sharding_spec.dim],  # type: ignore[union-attr, arg-type, index]
-        local_metadata.shard_sizes[sharding_spec.dim],  # type: ignore[union-attr, index]
-    ).contiguous()
+    # Narrow op is not included in the grad graph.
```

Comment:
nit: This comment might be more clear: "We don't want autograd recording here for the narrow op and 'local_shard' should be a leaf variable in the autograd graph".